packages/firebase/index.d.ts

```diff
@@ -7030,17 +7097,101 @@ declare namespace firebase.messaging {
     setBackgroundMessageHandler(
       callback: (payload: any) => Promise<any> | void
     ): void;
+
     /**
      * To use your own service worker for receiving push messages, you
      * can pass in your service worker registration in this method.
      *
      * @param registration The service worker
      *   registration you wish to use for push messaging.
+     * 
+     *     @deprecated Use getToken(options?: {
+      vapidKey?: string;
+      serviceWorkerRegistration?: ServiceWorkerRegistration;
+    }): Promise<string>;.
      */
+
     useServiceWorker(registration: ServiceWorkerRegistration): void;
+
+    /**
+     * @deprecated Use getToken(options?: {
+      vapidKey?: string;
+      serviceWorkerRegistration?: ServiceWorkerRegistration;
+    }): Promise<string>;.
+     */
     usePublicVapidKey(b64PublicKey: string): void;
   }
 
+  /**
+   * Message payload that contains the notification payload that is represented with
+   *  {@link firebase.Messaging.NotificationPayload} and the data payload that contains an
+   * arbitrary number of key-value pairs sent by developers through
+   * the {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notification Send API}
+   */
+  export interface MessagePayload {
+    /**
+     * See {@link firebase.Messaging.NotificationPayload}.
+     */
+    notification?: NotificationPayload;
+
+    /**
+     * Arbitrary key/value pairs.
+     */
+    data?: { [key: string]: string };
+
+    /**
+     * See {@link firebase.Messaging.FcmOptions}.
+     */
+    fcmOptions?: FcmOptions;
+
+    /**
+     * The sender of this message.
+     */
+    from: string;
+
+    /**
+     * The collapse key of this message. See more {@link https://firebase.google.com/docs/cloud-messaging/concept-options#collapsible_and_non-collapsible_messages}.
+     */
+    collapseKey: string;
+  }
+
+  /**
+   * Options for features provided by the FCM SDK for Web. See more {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#webpushfcmoptions}.
+   */
+  export interface FcmOptions {
+    /**
+     * The link to open when the user clicks on the notification. For all URL values, HTTPS is required.
+     * For example, by setting this value to your app's URL, a notification click click event will
+     * put your app in focus for the user.
+     */
+    link?: string;
+
+    /**
+     * Label associated with the message's analytics data.
```

Comment:
Do you think an @link would be good here?  Your call.  It could be https://firebase.google.com/docs/cloud-messaging/understand-delivery#adding-analytics-labels-to-messages.

Reply:
yay. would be nice to link a @link here